crates/sui-data-ingestion-core/src/reader.rs

```diff
@@ -257,16 +124,17 @@ impl CheckpointReader {
         .await?;
 
         let mut read_source: &str = "local";
-        if self.remote_store_url.is_some()
-            && (checkpoints.is_empty()
+        if let Some(remote_fetcher) = &mut self.remote_fetcher {
+            if checkpoints.is_empty()
                 || checkpoints[0].checkpoint_summary.sequence_number
-                    > self.current_checkpoint_number)
-        {
-            checkpoints = self.remote_fetch();
```

Comment:
> It does apply one optimization: we no longer re-create the remote object store every time we need to receive

Can you elaborate a bit? It wasn't created on every remote sync. The first time, it was created on demand. If the flow still prefers the remote fetch over a local fetch, the same remote object was used.
Or do you mean for the cases where you go back and forth between colocated and remote fetches?